.travis.yml

```diff
@@ -3,18 +3,20 @@ language: ruby
 cache:
   bundler: true
   directories:
-    - .phantomjs
+    - node_modules
     - vendor
     - .rvm
 
 services: postgresql
 
 addons:
+  chrome: stable
   postgresql: "9.4"
 
 before_script:
   - bundle exec rails db:create
   - bundle exec rails db:schema:load
+  - export DISPLAY=:99.0 && sh -e /etc/init.d/xvfb start && sleep 3
```

Comment:
You can also try using `xvfb-run` instead, i.e. `xvfb-run bundle exec rails test` in the `script:` section below. This starts Xvfb, sets `$DISPLAY`, runs the command when Xvfb is ready, and kills Xvfb again when the command has finished.

Reply:
Hmm actually, do we really need Xvfb? I haven't used Chrome Headless yet, but everything I can find about it so far seems to indicate that it's indeed truly headless, i.e. it shouldn't need an X server at all.

Reply:
I've simply taken examples from the net. They where using it, but of course I can try and see if it runs without

Reply:
I guess in some or another form xvfb is needed. I'm just guessing: Maybe it is needed for the screenshots?

For example FutureKids uses chrome headless since ages, and there xvfb is also run: https://github.com/panterch/future_kids/blob/master/.travis.yml

Reply:
There's some confusion with the terms:

- `chromedriver` is a version of Chrome that can be remotely controlled through the WebDriver API (using Selenium or others) and requires Xvfb (this is what FutureKids uses)
- `chrome --headless` is a new feature in Chrome 59 which supports running Chrome without Xvfb (this is what we're now using)

I tried to find out more and found this on https://developers.google.com/web/updates/2017/04/headless-chrome:

> **So I still need Xvfb?**
>
> No. Headless Chrome doesn't use a window so a display server like Xvfb is no longer needed. You can happily run your automated tests without it.
>
> What is Xvfb? Xvfb is an in-memory display server for Unix-like systems that enables you to run graphical applications (like Chrome) without an attached physical display. Many people use Xvfb to run earlier versions of Chrome to do "headless" testing.

Also noticed this question above:

> **Do I need the --disable-gpu flag?**
>
> Yes, for now. The --disable-gpu flag is a temporary requirement to work around a few bugs. You won't need this flag in future versions of Chrome. See https://crbug.com/546953#c152 and https://crbug.com/695212 for more information.

So I guess that's still needed for now :)